src/components/Account/ResetPassword/ResetPassword.component.js

```diff
@@ -18,103 +18,104 @@ import { forgot } from './ResetPassword.actions';
-    });
+  const handleSubmit = values => {
+    setIsSending(true);
+    setForgotState({});
 
     forgot(values)
-      .then(res => this.setState({ forgotState: res }))
-      .catch(err => this.setState({ forgotState: err }))
-      .finally(() => this.setState({ isSending: false }));
+      .then(res => setForgotState(res))
+      .catch(err => setForgotState(err))
+      .finally(() => setIsSending(false));
   };
 
-  render() {
-    const { isSending, forgotState } = this.state;
-    const { intl, isDialogOpen, onClose } = this.props;
-
-    const isButtonDisabled = isSending || !!forgotState.success;
+  const isButtonDisabled = isSending || !!forgotState.success;
 
-    return (
-      <Dialog open={isDialogOpen} onClose={onClose} aria-labelledby="forgot">
-        <DialogTitle id="forgot">
-          <FormattedMessage {...messages.resetPassword} />
-        </DialogTitle>
-        <DialogContent>
-          <DialogContentText>
-            <FormattedMessage {...messages.resetPasswordText} />
-          </DialogContentText>
+  return (
+    <Dialog open={isDialogOpen} onClose={onClose} aria-labelledby="forgot">
+      <DialogTitle id="forgot">
+        <FormattedMessage {...messages.resetPassword} />
+      </DialogTitle>
+      <DialogContent>
+        <DialogContentText>
+          <FormattedMessage {...messages.resetPasswordText} />
+        </DialogContentText>
 
-          <div
-            className={classNames('Forgot__status', {
-              'Forgot__status--error': !forgotState.success,
-              'Forgot__status--success': forgotState.success
-            })}
+        <div
+          className={classNames('Forgot__status', {
+            'Forgot__status--error': !forgotState.success,
+            'Forgot__status--success': forgotState.success
+          })}
+        >
+          {!!forgotState.success ? (
+            <Typography color="inherit">
+              {intl.formatMessage(messages.resetPasswordSuccess)}
+            </Typography>
+          ) : (
+            <Typography color="inherit">{forgotState.message}</Typography>
+          )}
+        </div>
+        {forgotState && !forgotState.success && (
+          <Formik
+            initialValues={{ email: '' }}
+            onSubmit={handleSubmit}
+            validationSchema={validationSchema}
+            enableReinitialize
           >
-            {!!forgotState.success ? (
-              <Typography color="inherit">
-                {intl.formatMessage(messages.resetPasswordSuccess)}
-              </Typography>
-            ) : (
-              <Typography color="inherit">{forgotState.message}</Typography>
+            {({ values, errors, handleChange, handleSubmit }) => (
+              <form className="Forgot__form" onSubmit={handleSubmit}>
+                <TextField
+                  error={errors.email}
+                  label={intl.formatMessage(messages.email)}
+                  name="email"
+                  value={values.email}
```

Comment:
You don't need to manually set the value property on the input, since <Formik> automatically controls and updates the form values based on its internal state.